ytmusicapi/auth/oauth.py

```diff
@@ -30,13 +30,35 @@ def is_custom_oauth(headers: CaseInsensitiveDict) -> bool:
 class YTMusicOAuth:
     """OAuth implementation for YouTube Music based on YouTube TV"""
 
-    def __init__(self, session: requests.Session, proxies: Dict = None):
+    def __init__(self,
+                 session: requests.Session,
+                 proxies: Dict = None,
+                 *,
+                 oauth_client_id: Optional[str] = None,
+                 oauth_client_secret: Optional[str] = None):
+        """
+        :param session: Session instance used for authorization requests.
+        :param proxies: Optional. Proxy configuration to be used by the session.
+        :param oauth_client_id: Optional. Keyword Only. Can be supplied to change the oauth client for token
+            retrieval and refreshing. Requires oauth_client_secret also be provided.
+        :param oauth_client_secret: Optional. Keyword Only. Used in conjunction with oauth_client_id to switch
+            underlying oauth client.
+        """
+
+        if not isinstance(oauth_client_id, type(oauth_client_secret)):
+            raise KeyError(
+                'Alternate oauth credential usage requires an id AND a secret. Pass both or neither.'
+            )
+
+        self.oauth_id = oauth_client_id if oauth_client_id else OAUTH_CLIENT_ID
+        self.oauth_secret = oauth_client_secret if oauth_client_secret else OAUTH_CLIENT_SECRET
```

Comment:
Was thinking about this overnight, and i think it's overcomplicated approach – instead of introducing new parameters(that has to be always kept in mind & carried over everywhere), why not make `OAUTH_CLIENT_ID` & `OAUTH_CLIENT_SECRET` to load themselves from env variables with fallback to default one's that @sigma67 dedicated for this project & hardcoded in `constants.py`?

Reply:
While I definitely agree on its complexity, I mentioned pursuing environmental variables briefly in the original issue #487 under alternative 1b. IMO, if the feature is something worth implementing, its worth implementing in a way that fully realizes its push for flexibility. My core disagreement with the env approach is that it hampers your ability to switch between clients within a process and also lacks clarity as to which client is currently active. 

That being said, I still like and agree with the utility that an option for using environmental variables provides. What are your thoughts on implementing them as an override within the dataclass mentioned above? IE the OAuthCredentials dataclass default values would be the environmental variables falling back to the dedicated constants when unavailable.

Reply:
Let's not start using environment variables :)